boringtun/src/sleepyinstant/mod.rs

```diff
@@ -0,0 +1,77 @@
+#![forbid(unsafe_code)]
+//! Attempts to provide the same functionality as std::time::Instant, except it
+//! uses a timer which accounts for time when the system is asleep
+use std::time::Duration;
+
+#[cfg(target_os = "windows")]
+mod windows;
+#[cfg(target_os = "windows")]
+use windows as inner;
+
+#[cfg(unix)]
+mod unix;
+#[cfg(unix)]
+use unix as inner;
+
+/// A measurement of a monotonically nondecreasing clock.
+/// Opaque and useful only with [`Duration`].
+///
+/// Instants are always guaranteed, barring [platform bugs], to be no less than any previously
+/// measured instant when created, and are often useful for tasks such as measuring
+/// benchmarks or timing how long an operation takes.
+///
+/// Note, however, that instants are **not** guaranteed to be **steady**. In other
+/// words, each tick of the underlying clock might not be the same length (e.g.
+/// some seconds may be longer than others). An instant may jump forwards or
+/// experience time dilation (slow down or speed up), but it will never go
+/// backwards.
+///
+/// Instants are opaque types that can only be compared to one another. There is
+/// no method to get "the number of seconds" from an instant. Instead, it only
+/// allows measuring the duration between two instants (or comparing two
+/// instants).
+///
+/// The size of an `Instant` struct may vary depending on the target operating
+/// system.
+///
+#[derive(Clone, Copy, Debug)]
+pub struct Instant {
```

Comment:
If we aren't publicly exposing this we should just make everything in here `pub(crate)`

Reply:
I left it as `pub` to make it easier to pull out into it's own crate in the future. I think that should be find since the module `sleepyinstant` is only exposed to the crate. Is there any harm in living this just as `pub`?

Reply:
This should be fine as the module isn't public, and it does make it easier to migrate later. I vote we leave it `pub`.

Reply:
Fair enough then.